### PR TITLE
[TOOLS-2916] Digest Section generator and necessary changes implemented

### PR DESCRIPTION
**Pull request by emannuelalmeida**

### CODE REVIEW

**Comment by objective-solutions-user** [comment]
SonarQube analysis reported 4 issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![BLOCKER][BLOCKER] [StrategicalProjectDataSet.java#L55](https://github.com/objective-solutions/taskboard/blob/ec81bf5c7f5ca6c1dd307c82e406f2c4202add66/dashboard-scenario/src/main/java/objective/taskboard/monitor/StrategicalProjectDataSet.java#L55): Rename field "name" to prevent any misunderstanding/clash with method "name" defined in superclass "java.lang.Enum". [rule](http://10.44.1.250:9000/coding_rules#rule_key=squid%3AS1845)
1. ![MAJOR][MAJOR] [ChartServiceUtils.java#L14](https://github.com/objective-solutions/taskboard/blob/ec81bf5c7f5ca6c1dd307c82e406f2c4202add66/dashboard-scenario/src/main/java/objective/taskboard/monitor/ChartServiceUtils.java#L14): Add a private constructor to hide the implicit public one. [rule](http://10.44.1.250:9000/coding_rules#rule_key=squid%3AS1118)
1. ![MAJOR][MAJOR] [BudgetProjectScenarioDSL.java#L336](https://github.com/objective-solutions/taskboard/blob/ec81bf5c7f5ca6c1dd307c82e406f2c4202add66/dashboard-scenario/src/test/java/objective/taskboard/monitor/BudgetProjectScenarioDSL.java#L336): Swap these 2 arguments so they are in the correct order: expected value, actual value. [rule](http://10.44.1.250:9000/coding_rules#rule_key=squid%3AS3415)
1. ![MAJOR][MAJOR] [BudgetProjectScenarioDSL.java#L343](https://github.com/objective-solutions/taskboard/blob/ec81bf5c7f5ca6c1dd307c82e406f2c4202add66/dashboard-scenario/src/test/java/objective/taskboard/monitor/BudgetProjectScenarioDSL.java#L343): Swap these 2 arguments so they are in the correct order: expected value, actual value. [rule](http://10.44.1.250:9000/coding_rules#rule_key=squid%3AS3415)

[BLOCKER]: https://sonarsource.github.io/sonar-github/severity-blocker.png 'Severity: BLOCKER'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'

File: dashboard-scenario/src/main/java/objective/taskboard/scenario/model/LogbookModel.java [hunk 1]
```diff
@@ -34,18 +34,22 @@ public TimelineParametersModel timelineParameters() {
         return timelineParameters;
     }
 
+    public BudgetParametersModel budgetParameters() { return budgetParametersModel; }
```
Comment (herbert-venancio) [inline dashboard-scenario/src/main/java/objective/taskboard/scenario/model/LogbookModel.java]: Suggestion: use new lines
```Java
    public BudgetParametersModel budgetParameters() {
        return budgetParametersModel;
    }
```

File: dashboard-scenario/src/main/java/objective/taskboard/scenario/model/ScenarioModel.java [hunk 2]
```diff
@@ -314,4 +327,23 @@ private TimelineParameters calculateTimelineParameters(LocalDate weekDate) {
...
+
+        budget.setConservativeBudget(getLastMapEntry(data.worklogConservativeProjection));
+        budget.setOptimisticBudget(getLastMapEntry(data.worklogOptimisticProjection));
+
+        return budget;
+    }
+
+    private double getLastMapEntry(Map<LocalDate, Double> series){
+        return series.get(series.keySet().stream()
+                .min(reverseOrder())
```
Comment (herbert-venancio) [inline dashboard-scenario/src/main/java/objective/taskboard/scenario/model/ScenarioModel.java]: Suggestion:
```Java
        return series.entrySet().stream()
                .max(Comparator.comparing(Map.Entry::getKey))
                .map(Map.Entry::getValue)
                .orElseThrow(IllegalStateException::new);
```
I don't know if it makes any difference in terms of performance

File: strategical-dashboard/src/main/java/objective/taskboard/monitor/DigestSectionGenerator.java [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+        );
+
+        return new LogbookSectionTableDto(1, "Digest", rows);
+    }
+
+    private String formatDate(LocalDate date) {
+        return date.format(formatter);
+    }
+
+    private String formatHours(double hours) {
```
Comment (herbert-venancio) [inline strategical-dashboard/src/main/java/objective/taskboard/monitor/DigestSectionGenerator.java]: `objective.taskboard.monitor.LogbookSectionUtils.formatHour` can't this be reused instead?

**Comment by objective-solutions-user** [comment]
SonarQube analysis reported 4 issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![BLOCKER][BLOCKER] [StrategicalProjectDataSet.java#L55](https://github.com/objective-solutions/taskboard/blob/21bf6bf7c2adbd77a826826769143f664f78077e/dashboard-scenario/src/main/java/objective/taskboard/monitor/StrategicalProjectDataSet.java#L55): Rename field "name" to prevent any misunderstanding/clash with method "name" defined in superclass "java.lang.Enum". [rule](http://10.44.1.250:9000/coding_rules#rule_key=squid%3AS1845)
1. ![MAJOR][MAJOR] [ChartServiceUtils.java#L14](https://github.com/objective-solutions/taskboard/blob/21bf6bf7c2adbd77a826826769143f664f78077e/dashboard-scenario/src/main/java/objective/taskboard/monitor/ChartServiceUtils.java#L14): Add a private constructor to hide the implicit public one. [rule](http://10.44.1.250:9000/coding_rules#rule_key=squid%3AS1118)
1. ![MAJOR][MAJOR] [BudgetProjectScenarioDSL.java#L336](https://github.com/objective-solutions/taskboard/blob/21bf6bf7c2adbd77a826826769143f664f78077e/dashboard-scenario/src/test/java/objective/taskboard/monitor/BudgetProjectScenarioDSL.java#L336): Swap these 2 arguments so they are in the correct order: expected value, actual value. [rule](http://10.44.1.250:9000/coding_rules#rule_key=squid%3AS3415)
1. ![MAJOR][MAJOR] [BudgetProjectScenarioDSL.java#L343](https://github.com/objective-solutions/taskboard/blob/21bf6bf7c2adbd77a826826769143f664f78077e/dashboard-scenario/src/test/java/objective/taskboard/monitor/BudgetProjectScenarioDSL.java#L343): Swap these 2 arguments so they are in the correct order: expected value, actual value. [rule](http://10.44.1.250:9000/coding_rules#rule_key=squid%3AS3415)

[BLOCKER]: https://sonarsource.github.io/sonar-github/severity-blocker.png 'Severity: BLOCKER'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'

**Pull request merged**